### DISCUSSION TITLE
useLoaderData returns a promise. Cannot display data.

### DISCUSSION THREAD

**Discussion by erAnusriM** [remix-run/react-router#10146] [General]
[](https://github.com/erAnusriM/react_router_trial)
https://github.com/erAnusriM/react_router_trial
The above is my github repository. 
How do I achieve to print the list of users returned from the API in App.jsx file
When I print the value of result on console, it gives me a Response object.

![image](https://user-images.githubusercontent.com/97402672/222116847-78a13404-6908-4ae8-b3c7-247b14d0f8e1.png)

**Comment by brophdawg11** [comment]
You need to unwrap the response from `fetch` in your loader via `response.json()`.  Defer works with promises and returns whatever you resolve the promise with, which is why you're getting a `Response` instance:

```js
function loader() { 
  const data = fetch(url).then(r => r.json());
  return defer({
    results: data,
  });
}
```
